packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -3213,6 +3258,118 @@ function flushRenderPhaseStrictModeWarningsInDEV() {
+      if (hasPassiveEffects) {
+        invokePassiveEffectsMountInDEV(fiber);
+      }
+    }
+  }
+}
+
+function invokePassiveEffectsMountInDEV(firstChild): void {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      let fiber = firstChild;
+      while (fiber !== null) {
+        if (fiber.child !== null) {
+          const primarySubtreeTag =
+            fiber.subtreeTag & MountPassiveDevSubtreeTag;
+          if (primarySubtreeTag !== NoSubtreeTag) {
+            invokePassiveEffectsMountInDEV(fiber.child);
+          }
+        }
+
+        const current = fiber.alternate;
+        if (current === null && (fiber.effectTag & Passive) !== NoEffect) {
+          invokeEffectMountInDEV(fiber, HookPassive | HookHasEffect);
+        }
+        fiber = fiber.sibling;
+      }
+    }
+  }
+}
+
+function invokePassiveEffectsUnmountInDEV(firstChild): void {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      let fiber = firstChild;
+      while (fiber !== null) {
+        if (fiber.child !== null) {
+          const primarySubtreeTag =
+            fiber.subtreeTag & MountPassiveDevSubtreeTag;
+          if (primarySubtreeTag !== NoSubtreeTag) {
+            invokePassiveEffectsUnmountInDEV(fiber.child);
+          }
+        }
+
+        const current = fiber.alternate;
+        if (current === null && (fiber.effectTag & Passive) !== NoEffect) {
+          invokeEffectUnmountInDEV(fiber, HookPassive | HookHasEffect);
+        }
+        fiber = fiber.sibling;
+      }
+    }
+  }
+}
+
+function invokeLayoutEffectsMountInDEV(firstChild) {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      let fiber = firstChild;
+      while (fiber !== null) {
+        if (fiber.child !== null) {
+          const primarySubtreeTag = fiber.subtreeTag & MountLayoutDevSubtreeTag;
+          if (primarySubtreeTag !== NoSubtreeTag) {
+            invokeLayoutEffectsMountInDEV(fiber.child);
+          }
+        }
+
+        const effectTag = fiber.effectTag;
+        const current = fiber.alternate;
+        if (current === null && (effectTag & Update) !== NoEffect) {
+          invokeEffectMountInDEV(fiber, HookLayout | HookHasEffect);
+        }
+        fiber = fiber.sibling;
+      }
+    }
+  }
+}
+
+function invokeLayoutEffectsUnmountInDEV(firstChild) {
+  if (enableDoubleInvokingEffects) {
+    if (__DEV__) {
+      let fiber = firstChild;
+      while (fiber !== null) {
+        if (fiber.child !== null) {
+          const primarySubtreeTag = fiber.subtreeTag & MountLayoutDevSubtreeTag;
+          if (primarySubtreeTag !== NoSubtreeTag) {
+            invokeLayoutEffectsUnmountInDEV(fiber.child);
+          }
+        }
+
+        const effectTag = fiber.effectTag;
+        const current = fiber.alternate;
+
+        if (current === null && (effectTag & Update) !== NoEffect) {
+          invokeEffectUnmountInDEV(fiber, HookLayout | HookHasEffect);
+        }
+        fiber = fiber.sibling;
+      }
+    }
+  }
+}
```

Comment:
IMO it would be nice to DRY `invokePassiveEffectsMountInDEV`/`invokeLayoutEffectsMountInDEV` and `invokePassiveEffectsUnmountInDEV`/`invokeLayoutEffectsUnmountInDEV` and just pass in a tag (`HookPassive | HookHasEffect` vs `HookLayout | HookHasEffect`). We're already doing this for `invokeEffectMountInDEV` and `invokeEffectUnmountInDEV` too. :)